src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests2.cs

```diff
@@ -2596,5 +2596,197 @@ public void M(string s)
+            {
+                var @switch = switches[i];
+                var typeInfo = model.GetTypeInfo(@switch);
+                var conversion = model.GetConversion(@switch);
+                switch (i)
+                {
+                    case 0:
+                    case 7:
+                        Assert.Null(typeInfo.Type);
+                        Assert.Equal("C", typeInfo.ConvertedType.ToTestDisplayString());
+                        Assert.Equal(ConversionKind.SwitchExpression, conversion.Kind);
+                        break;
+                    case 1:
+                        Assert.Null(typeInfo.Type);
+                        Assert.True(typeInfo.ConvertedType.IsErrorType());
+                        Assert.Equal(ConversionKind.Identity, conversion.Kind);
+                        break;
+                    case 2:
+                    case 8:
+                        Assert.Null(typeInfo.Type);
+                        Assert.Equal("D", typeInfo.ConvertedType.ToTestDisplayString());
+                        Assert.Equal(ConversionKind.SwitchExpression, conversion.Kind);
+                        break;
+                    case 3:
+                        Assert.True(typeInfo.Type.IsErrorType());
+                        Assert.Equal("D", typeInfo.ConvertedType.ToTestDisplayString());
+                        Assert.Equal(ConversionKind.NoConversion, conversion.Kind);
+                        break;
+                    case 9:
+                        Assert.Null(typeInfo.Type);
+                        Assert.Equal("?", typeInfo.ConvertedType.ToTestDisplayString());
+                        Assert.Equal(ConversionKind.Identity, conversion.Kind);
+                        break;
+                    case 4:
+                    case 10:
+                        Assert.Equal("C", typeInfo.Type.ToTestDisplayString());
+                        Assert.Equal("C", typeInfo.ConvertedType.ToTestDisplayString());
+                        Assert.Equal(ConversionKind.Identity, conversion.Kind);
+                        break;
+                    case 5:
+                        Assert.Equal("C", typeInfo.Type.ToTestDisplayString());
+                        Assert.Equal("D", typeInfo.ConvertedType.ToTestDisplayString());
+                        Assert.Equal(ConversionKind.ImplicitUserDefined, conversion.Kind);
+                        break;
+                    case 11:
+                        Assert.Equal("C", typeInfo.Type.ToTestDisplayString());
+                        Assert.Equal("C", typeInfo.ConvertedType.ToTestDisplayString());
+                        Assert.Equal(ConversionKind.Identity, conversion.Kind);
+                        break;
+                    case 6:
+                    case 12:
+                        Assert.Equal("System.Int32", typeInfo.Type.ToTestDisplayString());
+                        Assert.Equal("D", typeInfo.ConvertedType.ToTestDisplayString());
```

Comment:
>Assert.Equal("D", typeInfo.ConvertedType.ToTestDisplayString()); [](start = 24, length = 64)

This looks unexpected for the explicit cast case. The ```ConvertedType``` is supposed to reflect an implicit conversion information rather than explicit conversion information. For example, an integer literal converted to something explicitly has both types as ```int```.

Reply:
The explicit conversion is preferred over an implicit switch expression conversion.

---
In reply to: [440449755](https://github.com/dotnet/roslyn/pull/45069#discussion_r440449755) [](ancestors = 440449755)

Reply:
> The explicit conversion is preferred over an implicit switch expression conversion for a cast.

```ConvertedType``` never reflects explicit conversions

---
In reply to: [440450634](https://github.com/dotnet/roslyn/pull/45069#discussion_r440450634) [](ancestors = 440450634,440449755)

Reply:
This is about conversion explicitly stated in code versus implicitly applied by compiler. It is not about conversion classification (explicit vs. implicit)

---
In reply to: [440451631](https://github.com/dotnet/roslyn/pull/45069#discussion_r440451631) [](ancestors = 440451631,440450634,440449755)

Reply:
The conversion explicitly stated in the code is treated as an identity conversion. The underlying conversion, which is performed by the switch expression itself, is a switch expression conversion. That is what is reported here.

---
In reply to: [440453038](https://github.com/dotnet/roslyn/pull/45069#discussion_r440453038) [](ancestors = 440453038,440451631,440450634,440449755)